### PR TITLE
chore: update engines to 3.5.0-20.00ae542f0021d0d0a22b970a9ae56a6b170d5fd0

### PR DESCRIPTION
**Pull request by aqrln**
Refs https://github.com/prisma/prisma/issues/9416
Refs https://github.com/prisma/prisma-engines/pull/2390

### CODE REVIEW

File: packages/client/src/__tests__/types/selectRelationCount/test.ts [hunk 1]
```diff
@@ -12,8 +12,8 @@ async function main() {
     },
   })
 
-  const postsCount: number | undefined = users[0]._count?.posts
-  const likeCount: number | undefined = users[0]._count?.Like
+  const postsCount: number | undefined = users[0]._count.posts
```
Comment (millsp) [inline packages/client/src/__tests__/types/selectRelationCount/test.ts:15]: is `undefined` still needed here?

**Review by millsp — APPROVED** [review]

File: packages/client/src/__tests__/types/selectRelationCount/test.ts (hunk 1 above)
Comment (aqrln) [reply packages/client/src/__tests__/types/selectRelationCount/test.ts:15]: Definitely not, thanks!

**Comment by aqrln** [comment]
Blocked by https://github.com/prisma/prisma/pull/10180

**Review by Jolg42 — APPROVED** [review]
Looks good! I think you can merge after resolving the conflict due to another engine bump PR merged

**Comment by aqrln** [comment]
A flaky test?

```
[2021-11-10T17:04:40Z]  FAIL  src/__tests__/integration/happy/disconnect-race/test.ts
--
  | [2021-11-10T17:04:40Z]   ● disconnect-race
  | [2021-11-10T17:04:40Z]
  | [2021-11-10T17:04:40Z]
  | [2021-11-10T17:04:40Z]     Invalid `prisma.user.findMany()` invocation in
  | [2021-11-10T17:04:40Z]     /app/packages/client/src/__tests__/integration/happy/disconnect-race/test.ts:9:31
  | [2021-11-10T17:04:40Z]
  | [2021-11-10T17:04:40Z]        6
  | [2021-11-10T17:04:40Z]        7 await prisma.user.findMany()
  | [2021-11-10T17:04:40Z]        8 prisma.$disconnect()
  | [2021-11-10T17:04:40Z]     →  9 const a = await prisma.user.findMany(
  | [2021-11-10T17:04:40Z]       Engine is not yet connected.
  | [2021-11-10T17:04:40Z]        0: user_facing_errors::Error::new_non_panic_with_current_backtrace
  | [2021-11-10T17:04:40Z]        1: query_engine::error::<impl core::convert::From<query_engine::error::ApiError> for napi::error::Error>::from
  | [2021-11-10T17:04:40Z]        2: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
  | [2021-11-10T17:04:40Z]        3: tokio::runtime::task::harness::Harness<T,S>::poll
  | [2021-11-10T17:04:40Z]        4: tokio::runtime::thread_pool::worker::Context::run_task
  | [2021-11-10T17:04:40Z]        5: tokio::runtime::thread_pool::worker::Context::run
  | [2021-11-10T17:04:40Z]        6: tokio::macros::scoped_tls::ScopedKey<T>::set
  | [2021-11-10T17:04:40Z]        7: tokio::runtime::thread_pool::worker::run
  | [2021-11-10T17:04:40Z]        8: tokio::loom::std::unsafe_cell::UnsafeCell<T>::with_mut
  | [2021-11-10T17:04:40Z]        9: tokio::runtime::task::harness::Harness<T,S>::poll
  | [2021-11-10T17:04:40Z]       10: tokio::runtime::blocking::pool::Inner::run
  | [2021-11-10T17:04:40Z]       11: std::sys_common::backtrace::__rust_begin_short_backtrace
  | [2021-11-10T17:04:40Z]       12: core::ops::function::FnOnce::call_once{{vtable.shim}}
  | [2021-11-10T17:04:40Z]       13: <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once
  | [2021-11-10T17:04:40Z]
  | [2021-11-10T17:04:40Z]       at ../../../rustc/c8dfcfe046a7680554bf4eb612bad840e7631c4b/library/alloc/src/boxed.rs:1572:9
  | [2021-11-10T17:04:40Z]             <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once
  | [2021-11-10T17:04:40Z]       at ../../../rustc/c8dfcfe046a7680554bf4eb612bad840e7631c4b/library/alloc/src/boxed.rs:1572:9
  | [2021-11-10T17:04:40Z]             std::sys::unix::thread::Thread::new::thread_start
  | [2021-11-10T17:04:40Z]       at ../../../rustc/c8dfcfe046a7680554bf4eb612bad840e7631c4b/library/std/src/sys/unix/thread.rs:74:17
  | [2021-11-10T17:04:40Z]         14: start_thread
  | [2021-11-10T17:04:40Z]         15: clone
  | [2021-11-10T17:04:40Z]       at cb (src/runtime/PrismaClientFetcher.ts:179:17)
  | [2021-11-10T17:04:40Z]       at Object.<anonymous> (src/__tests__/integration/happy/disconnect-race/test.ts:9:13)
  | [2021-11-10T17:04:40Z]
  | [2021-11-10T17:04:40Z]
```

**Comment by Jolg42** [comment]
Looks flaky indeed!

**Pull request merged**